### PR TITLE
Added options tab, added user setting points per win/tie/loss

### PR DESCRIPTION
**Pull request by Leothechosen**
Adds an options tab, as well as letting the user set their tournament's points per win/tie/loss

Changes:

1. default-tournament.json: 
    - Added `points_on_win_tie_loss` field. Defaults to `[1, 0, 0]`, respectively.
2. form.ui: 
    - Added `Options` tab,
    - Added fields to change points on win/tie/loss. Only supports integers.
    - Added `Edit Points` button to push the changes to points on win/tie/loss
3. tourneydefs.py:
    - Tournament class now has `pts_on_win`, `pts_on_tie`, and `pts_on_loss` attributes.
    - `load_from()` now loads `points_on_win_tie_loss` from the tournament config file.
    - `save_to()` now saves `points_on_win_tie_loss` to the tournament config file.
    - `get_standings()` now uses the `pts_on_xxx` attributes when adding points to teams.
    - Created `edit_points()` to set the Tournament attributes to new values set by user.
4. udts.py:
    - `setup()` now has a handler for when `Edit Points` is clicked.
    - Added `edit_points()` to handle when `Edit Points` is clicked. 
    - Writing to `tournament-config.json` now also writes `points_on_win_loss_tie`

### CODE REVIEW

File: tourneydefs.py [hunk 1]
```diff
@@ -85,6 +85,9 @@ class Tournament(BaseModel):
     game_history: List[Game] = []
     mapping: Dict = {}
     version : str = "0.3"
+    pts_on_win: int = 0
+    pts_on_tie: int = 0
+    pts_on_loss: int = 0
```
Comment (chhopsky) [inline tourneydefs.py:90]: This feels like it should be a tournament configuration option - my rec here is have a dict property so we can auto-export/import any new config options that get added.

`config: Dict = {     pts_on_win: 1
    pts_on_tie: 0
    pts_on_loss: 0 }`

Also feel free to write getter/setter for this to avoid keyerrors and ease refactoring

File: tourneydefs.py [hunk 2]
```diff
@@ -115,6 +118,12 @@ def load_from(self, filename="tournament-config.json"):
                 current_match = data.get("current_match")
                 if current_match is not None:
                     self.current_match = current_match
+                
+                points_on_win_tie_loss = data["points_on_win_tie_loss"]
+                self.pts_on_win = points_on_win_tie_loss[0]
+                self.pts_on_tie = points_on_win_tie_loss[1]
+                self.pts_on_loss = points_on_win_tie_loss[2]
```
Comment (chhopsky) [inline tourneydefs.py:125]: as above, consolidate all tournament config into a single config dict or similar

File: udts.py [hunk 3]
```diff
@@ -210,6 +211,14 @@ def save_as_state():
...
 
 
+def edit_points():
+    pts_on_win = int(window.points_on_win_spinbox.text())
+    pts_on_tie = int(window.points_on_tie_spinbox.text())
+    pts_on_loss = int(window.points_on_loss_spinbox.text())
+    broadcast.edit_points([pts_on_win, pts_on_tie, pts_on_loss])
+    force_refresh_ui()
+
+
```
Comment (chhopsky) [inline udts.py:221]: should probably consolidate on how these should be stored and accessed. if list then let's use a list everywhere. ideally this should be un-fuck-up-able as possible. separate values? a configuration object that stores them separately? not sure, but probably not a great habit to get into of passing lists like this where position dictates interpretation

File: udts.py [hunk 4]
```diff
@@ -603,7 +612,8 @@ def save_config(config_to_save):
     config = { "openfile": "tournament-config.json",
         "use_challonge": False,
         "challonge_id": False,
-        "version": version
+        "version": version,
+        "points_on_win_tie_loss": [1, 0, 0]
```
Comment (chhopsky) [inline udts.py:616]: as above, this is a tournament configuration option that should be saved in tournament-config.json

**Review by chhopsky — CHANGES_REQUESTED** [review]
Some queries around lists-as-interpretation and where config should go

**Review by chhopsky — APPROVED** [review]
